strings/base_macros.h

```diff
@@ -118,39 +161,18 @@ namespace winrt::impl
             return m_file;
         }
 
-        constexpr const char* function_name() const noexcept
+        [[nodiscard]] constexpr const char* function_name() const noexcept
         {
-            // This is intentionally not included.  See comment above.
-            return nullptr;
+            return m_function;
         }
 
     private:
         const std::uint_least32_t m_line{};
         const char* const m_file{};
+        const char* const m_function{};
     };
 }
 
-// std::source_location includes function_name which can be helpful but creates a lot of binary size impact.  Many consumers
-// have defined WINRT_NO_SOURCE_LOCATION to prevent this impact, losing the value of source_location.  We have defined a
-// slim_source_location struct that is equivalent but excludes function_name.  This should have the vast majority of the
-// usefulness of source_location while having a much smaller binary impact.
-//
-// When building _DEBUG binary size is not usually much of a concern, so we can use the full source_location type.
-#ifdef _DEBUG
-#define WINRT_IMPL_SOURCE_LOCATION_ARGS_NO_DEFAULT , std::source_location const& sourceInformation
-#define WINRT_IMPL_SOURCE_LOCATION_ARGS , std::source_location const& sourceInformation = std::source_location::current()
-#define WINRT_IMPL_SOURCE_LOCATION_ARGS_SINGLE_PARAM std::source_location const& sourceInformation = std::source_location::current()
-
-#define WINRT_IMPL_SOURCE_LOCATION_FORWARD , sourceInformation
-#define WINRT_IMPL_SOURCE_LOCATION_FORWARD_SINGLE_PARAM sourceInformation
-
-#define WINRT_SOURCE_LOCATION_ACTIVE
-
-#ifdef _MSC_VER
-#pragma detect_mismatch("WINRT_SOURCE_LOCATION", "true")
-#endif // _MSC_VER
-
-#else // !_DEBUG
 #define WINRT_IMPL_SOURCE_LOCATION_ARGS_NO_DEFAULT , winrt::impl::slim_source_location const& sourceInformation
 #define WINRT_IMPL_SOURCE_LOCATION_ARGS , winrt::impl::slim_source_location const& sourceInformation = winrt::impl::slim_source_location::current()
 #define WINRT_IMPL_SOURCE_LOCATION_ARGS_SINGLE_PARAM winrt::impl::slim_source_location const& sourceInformation = winrt::impl::slim_source_location::current()
```

Comment:
Since this no longer evals to a no-op, we technically don't need the `_SINGLE_PARAM`, which only exists to remove the comma. We could eliminate that variant and explicitly supply the comma in the callsites. Not required, but something to consider.

Reply:
Minimizing churn sounds fine to me.

Reply:
I've never like these macros so I'd be in favor of removing them even if it causes a bit of churn.

Reply:
I never liked them either, even though I added them.  By popular request they are now gone :).